kf_lib_data_ingest/common/constants.py

```diff
@@ -399,28 +402,59 @@ class RELATIONSHIP:
     TWIN_BROTHER = "Twin Brother"
     TWIN_SISTER = "Twin Sister"
     TWIN = "Twin"
+    HALF_BROTHER = "Half-Brother"
+    HALF_SISTER = "Half-Sister"
+    HALF_SIBLING = "Half-Sibling"
```

Comment:
I think it would make sense to specify if the half-siblings are paternal or maternal. Please add Paternal and Maternal entries for each

Reply:
Fully agreed, thanks for the catch!  

In addition to the 4x other areas to add any such new relationships, should I also add them to:
5. the appropriate relationship-parent mappings ([here](https://github.com/kids-first/kf-lib-data-ingest/blob/123ae7b686b71f56abfe3907fe5d5fcff71e1ba4/kf_lib_data_ingest/common/constants.py#L556-L581))
???

Reply:
Scratch that about #5, since I now better see that is the same as #4.

Reply:
@chris-s-friedman , et al: for a genderless reverse relationship from Aunts and Uncles..... shall we use "Niblings"?
as per: https://www.merriam-webster.com/words-at-play/words-were-watching-nibling

Reply:
In discussing further with the KF-DRC team, wondering if "nibling" and "pibling" could be used as the default genderless versions (of niece/nephew, and aunt/uncle, specifically), with the gendered reversals defined to help refine these to the gender specific version.

Reply:
I am wondering if the reverse relationships _need_ to be genderless for nephew/niece and aunt/uncle. I believe when we say reverse relationship we're referring to populating the value of `participant2_to_participant1_relation` ? it would make sense for this to be left as niece/nephew or aunt/uncle... ?

Reply:
Well, at the first pass, prior to the kf-ingest-data-ingest bringing in the gender and refining it where possible, (much like Sibling is set as the opposite of both Brother and Sister, and Child as the opposite of both Father and Mother) this would allow the first-pass default of a genderless value, should gender not exist on the participant for whatever reason.

Reply:
Yea I guess I'm wondering in what circumstances would we have a participant without a sex and yet have an indicated relationship? And specifically wondering how that situation would arise with aunt/uncle and niece/nephew? If the submitted relationship is "niece", "nephew", "aunt", "uncle" the given sex of subject should align, and if it doesn't isnt this something we would bring up to the X01 group and have corrected?

Reply:
Thanks for the thoughts, @Christina-J-Diaz .  

Even if these particular participants in this clinical data set should all have their sex, it seems that this kf-lib-data-ingest system is generally setup for the genderless default to be the first pass value before it then performs any such gender refinement.  As such, it does feel to me that to not include something there (even if it's a new one to us, such as "nibling" and "pibling") as its first pass default might be remiss.  Doesn't adding these genderless reverse lookups align with the spirit of the kf-lib-data-ingest system?  Thanks!

Reply:
Another thought on this.... if the terms "nibling" and "pibling" in particular are of concern, what about just:
* "Niece or Nephew"
* "Aunt or Uncle"

or perhaps even a hybrid combination, such as:
* "Nibling (Niece or Nephew)"
* "Pibling (Aunt or Uncle)"
?

Reply:
For your consideration, I have committed potential refinements (https://github.com/kids-first/kf-lib-data-ingest/pull/657/commits/a09004acecf995df1c83b19904f8a989de26fb05) to demonstrate the possibilities of such hybrid combinations, e.g.,:
> PIBLING = "Pibling (Aunt or Uncle)"
> NIBLING = "Nibling (Niece or Nephew)"

Reply:
@r3m0chop Maybe we can use "other" for right now? Since this will be default behavior, it will affect all ingest packages--including other teams that may use the packages (like the M1 team). Maybe we can default to "other" temporarily and then have a larger discussion about this in a separate PR?

Reply:
Thank you, @Christina-J-Diaz .  

Howzabout the following for such initial "other" text description placeholders (noting the internal constant name shouldn't be visible outside of our code):

*     MATERNAL_PIBLING = "Maternal Other (Aunt or Uncle)"
*     PATERNAL_PIBLING = "Paternal Other (Aunt or Uncle)"
*     PIBLING = "Other (Aunt or Uncle)"
*     MATERNAL_NIBLING = "Maternal Other (Niece or Nephew)"
*     PATERNAL_NIBLING = "Paternal Other (Niece or Nephew)"
 *    NIBLING = "Other (Niece or Nephew)"
etc

Reply:
@r3m0chop I like this. I think it is a good compromise for a placeholder. I appreciate the additional "maternal" / "paternal" description with other-- that is more informative than leaving as "other". I'm not 100% sold on the parentheses (because its a much longer than string compared to the other values, but I don't see an issue with it for now 😄 )

Reply:
This might have an obvious answer but thinking about vocabulary in terms of pedigrees -- do we need the maternal / paternal distinction for niece / nephew? Biologically you only have one set of nieces/nephews , which are all the children of your sibling

Reply:
Great thoughts, folks. just to build on this, saying `Other (some stuff in parens)` is confusing - what about some other *other* relation besides aunt/uncle, niece/ nephew? 

How about: 

> MATERNAL_PIBLING = "Maternal Aunt or Uncle"
PATERNAL_PIBLING = "Paternal Aunt or Uncle"
PIBLING = "Aunt or Uncle"
MATERNAL_NIBLING = "Maternal Niece or Nephew"
PATERNAL_NIBLING = "Paternal Niece or Nephew"
NIBLING = "Niece or Nephew"

Reply:
Great point. Agreed -- this looks cleaner. @r3m0chop what do you think? I am actually appreciating that the constant name is PIBLING or NIBLING and the value is the _definition_ of pibling/nibling.